apps/passport-client/src/useSyncE2EEStorage.tsx

```diff
@@ -151,18 +160,50 @@ export async function uploadSerializedStorage(
       success: true,
       value: { revision: uploadResult.value.revision, storageHash: storageHash }
     };
+  } else if (uploadResult.error.name === "Conflict") {
+    console.warn("[SYNC] conflict uploading e2ee storage", uploadResult.error);
+    return { success: false, error: uploadResult.error };
   } else {
     console.error("[SYNC] failed to upload e2ee storage", uploadResult.error);
     return { success: false, error: uploadResult.error };
   }
 }
 
+export type MergeState = {
+  pcds: PCDCollection;
+  subscriptions: FeedSubscriptionManager;
+};
+
+export type MergeStorageResult = APIResult<MergeState, NamedAPIError>;
```

Comment:
It is an interesting choice to use `APIResult` in this way (this way being not strictly as a translation between the complexities of making an HTTP request over the network and the rest of the application; this way being as an encapsulation of a complicated operation that has several distinct failure modes). One concern I have about 'this way' is that it may result in us having to do an inconvenient refactor in the future, for example if we make a change to how API requests between the client and server are made. On the other hand, the concept of 'either a value or an error' is useful, and maybe `APIResult` should be promoted to a more generic thing by renaming it to something like `MaybeResult`. It is also worth considering whether or not we actually want to enshrine this concept in this way. Personally I'm not opposed, but also not 100% sure.

Reply:
Agreed that what I'm using here is an error-handling pattern which happens to be convenient, not strictly tied to HTTP APIs.  Its usage has propagated through the sync-related code because much of it actually does make HTTP requests.  I'm all for thinking about how to generify it properly, though I don't think that has to be in this PR.

I could as easily do this with exceptions, and might still switch to that later if it's convenient, though we also don't have a good pattern for distinguishing different error cases in exceptions.  If I wanted it, I might use something like NamedAPIError there too.